go.mod

```diff
@@ -1,3 +1,7 @@
 module go.yaml.in/yaml/v4
 
 go 1.18
+
+require go.yaml.in/yaml/v3 v3.0.4
+
+replace go.yaml.in/yaml/v3 => ./.cache/yaml-v3
```

Comment:
There is something really strange to me here

Having the v4 to depends on v3 🤔

Either things should be shared between version, and we need something like an external repository like yts, either there is an error here.

Reply:
Let me explain.

The tests are written in basic yaml and so they need a yaml decoder to decode them, in order to work. This could be any yaml loader, not even ours. 

Of course I wanted it to be ours. I couldn't use the current version 4 as it caused a circular dep. I didn't need v4 because this is just simple yaml where I need the data. I decided to use the most recent v3 of ours. It is perfectly fine.

I could have just depended on it but we don't have any deps for this library so that would be weird. Especially since we have all the code we need right here. So I used a replace and had the Makefile make sure it was on disk.

I find it a pretty great solution, but open to suggestions...